app/javascript/packs/pages/volunteer_signup/VolunteerStepper.jsx

```diff
@@ -0,0 +1,257 @@
+import StepLabel from "@material-ui/core/StepLabel";
+import Button from "@material-ui/core/Button";
+import Typography from "@material-ui/core/Typography";
+import { gql } from "apollo-boost";
+import { useMutation, useQuery } from "@apollo/react-hooks";
+import ErrorIcon from "@material-ui/icons/Error";
+import Snackbar from "@material-ui/core/Snackbar";
+import queryString from 'query-string';
+
+const useStyles = makeStyles(theme => ({
+  paper: {
+    marginTop: theme.spacing(3),
+    padding: theme.spacing(2),
+    [theme.breakpoints.up(600 + theme.spacing(3) * 2)]: {
+      marginTop: theme.spacing(6),
+      padding: theme.spacing(3)
+    }
+  },
+  stepper: {
+    padding: theme.spacing(5, 0, 3)
+  },
+  buttons: {
+    display: "flex",
+    justifyContent: "flex-end"
+  },
+  button: {
+    marginTop: theme.spacing(3),
+    marginLeft: theme.spacing(1)
+  },
+  error: {
+    backgroundColor: theme.palette.error.main,
+    color: theme.palette.error.contrastText,
+    fill: theme.palette.error.contrastText,
+    padding: theme.spacing(1)
+  },
+  step: {
+    cursor: "pointer"
+  },
+  heading: {
+    margin: theme.spacing(2)
+  },
+  details: {
+    fontWeight: 100,
+    fontSize: "1rem"
+  }
+}));
+
+const CREATE_VOLUNTEER = gql`
+  mutation CreateVolunteer(
+    $firstName: String!
+    $lastName: String
+    $neighborhood: String
+    $city: String!
+    $state: String!
+    $email: String!
+    $providerId: ID!
+    $requests: [String!]!
+    $description: String
+    $availabilities: [String!]!
+    $phone: String
+    $social: String
+    $over18: Boolean!
+  ) {
+    createVolunteer(
+      input: {
+        firstName: $firstName
+        lastName: $lastName
+        neighborhood: $neighborhood
+        city: $city
+        state: $state
+        email: $email
+        providerId: $providerId
+        requests: $requests
+        description: $description
+        availabilities: $availabilities
+        phone: $phone
+        social: $social
+        over18: $over18
+      }) {
+        errors
+        volunteer { id, firstName, responses { provider { id }, requests, availabilities } }
+      }
+  }
+`;
+
+const GET_PROVIDER = gql`
+  query Provider($id: ID!) {
+    provider(id: $id) {
+      edges {
+        node {
+          id
+          firstName
+          requests {
+            type,
+            satisfied
+          }
+        }
+      }
+    }
+  }
+`;
+
+export default function VolunteerStepper({ steps, location }) {
+  const classes = useStyles();
+  const [activeStep, setActiveStep] = React.useState(0);
+  const [errors, setErrors] = React.useState(null);
+  const [variables, setVariables] = React.useState({
+    firstName: "",
+    lastName: "",
+    neighborhood: "",
+    city: "",
+    state: "",
+    email: "",
+    requests: [],
+    description: "",
+    availabilities: [],
+    phone: "",
+    social: "",
+    over18: false
+  });
+
+  const queryId = queryString.parse(location.search).provider
```

Comment:
Super minor: `react-router-dom` has a hook `useQuery`. That would look like `const { id } = useQuery()`. No need to change just a heads up!

Reply:
@tuntisz thanks for the review!! I'm going to make this change before merging.

Reply:
It looks like react-router removed this capability in v4, which is unfortunate! I've definitely used it before on other projects, but now it makes sense why I wasn't seeing the option this time.

Reply:
Doh! Good to know.